### PR TITLE
Add support for passing ref for drone plugin by name

### PR DESCRIPTION
**Pull request by shubham149**

### CODE REVIEW

File: plugin/internal/file/file.go [hunk 1]
```diff
@@ -42,7 +45,7 @@ func download(url, path string) error {
 	}
 	resp, err := client.Get(url)
 	if err != nil {
-		return errors.Wrap(err, fmt.Sprintf("failed to download url: %s", url))
```
Comment (hemanthmantri) [inline plugin/internal/file/file.go:45]: why not print url?

File: plugin/harness/lookup.go [hunk 2]
```diff
@@ -43,7 +43,11 @@ func ParseLookup(s string) (repo string, commit string, ok bool) {
 		// extract the version from the string, if provided.
 		if parts := strings.SplitN(s, "@", 2); len(parts) == 2 {
 			repo = parts[0]
-			commit = parts[1]
+			if strings.HasPrefix(parts[1], "refs/") {
```
Comment (hemanthmantri) [inline plugin/harness/lookup.go:46]: Add an example pattern of string "s" in a comment.

File: plugin/internal/file/file.go (hunk 1 above)
Comment (shubham149) [reply plugin/internal/file/file.go:45]: Adding url in L28

File: plugin/harness/lookup.go (hunk 2 above)
Comment (shubham149) [reply plugin/harness/lookup.go:46]: Added UTs for it for example

**Pull request merged**